### PR TITLE
contrib/python-adblock: new package (0.6.0)

### PR DESCRIPTION
**Pull request by ttyyls**

### CODE REVIEW

**Comment by ttyyls** [comment]
split from #1828

File: contrib/python-adblock/template.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+makedepends = ["rust-std"]
+checkdepends = ["python-pytest-xdist"]
+pkgdesc = "Python wrapper for brave's adblocking library"
+maintainer = "ttyyls <contact@behri.org>"
+license = "Apache-2.0 OR MIT"
+url = "https://github.com/ArniDagur/python-adblock"
+source = f"{url}/archive/refs/tags/{pkgver}.tar.gz"
+sha256 = "cdde5db640f1d9205785641665b79e6edb7c51092a68536bc4907560c71136b2"
+# XXX: requires pre-installing the package
+options = ["!check"]
```
Comment (nekopsykose) [inline contrib/python-adblock/template.py:22]: the pep517 check does install the wheel so this should work, how does it fail?

Comment (ttyyls) [reply contrib/python-adblock/template.py:22]: throws some import errors so I thought it simply wasn't getting pre-installed correctly

```
=> python-adblock-0.6.0-r0: running do_check...
============================= test session starts ==============================
platform linux -- Python 3.12.2, pytest-8.2.1, pluggy-1.5.0
rootdir: /builddir/python-adblock-0.6.0
configfile: pyproject.toml
plugins: xdist-3.6.1
collected 1 item / 6 errors

==================================== ERRORS ====================================
____________________ ERROR collecting tests/test_engine.py _____________________
ImportError while importing test module '/builddir/python-adblock-0.6.0/tests/test_engine.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_engine.py:1: in <module>
    import adblock
adblock/__init__.py:1: in <module>
    from adblock.adblock import (
E   ModuleNotFoundError: No module named 'adblock.adblock'
__________________ ERROR collecting tests/test_exceptions.py ___________________
ImportError while importing test module '/builddir/python-adblock-0.6.0/tests/test_exceptions.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_exceptions.py:1: in <module>
    import adblock
adblock/__init__.py:1: in <module>
    from adblock.adblock import (
E   ModuleNotFoundError: No module named 'adblock.adblock'
____________________ ERROR collecting tests/test_imports.py ____________________
ImportError while importing test module '/builddir/python-adblock-0.6.0/tests/test_imports.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_imports.py:2: in <module>
    import adblock
adblock/__init__.py:1: in <module>
    from adblock.adblock import (
E   ModuleNotFoundError: No module named 'adblock.adblock'
___________________ ERROR collecting tests/test_metadata.py ____________________
ImportError while importing test module '/builddir/python-adblock-0.6.0/tests/test_metadata.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_metadata.py:5: in <module>
    import adblock
adblock/__init__.py:1: in <module>
    from adblock.adblock import (
E   ModuleNotFoundError: No module named 'adblock.adblock'
___________________ ERROR collecting tests/test_redirect.py ____________________
ImportError while importing test module '/builddir/python-adblock-0.6.0/tests/test_redirect.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_redirect.py:1: in <module>
    import adblock
adblock/__init__.py:1: in <module>
    from adblock.adblock import (
E   ModuleNotFoundError: No module named 'adblock.adblock'
_____________________ ERROR collecting tests/test_repr.py ______________________
ImportError while importing test module '/builddir/python-adblock-0.6.0/tests/test_repr.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_repr.py:1: in <module>
    import adblock
adblock/__init__.py:1: in <module>
    from adblock.adblock import (
E   ModuleNotFoundError: No module named 'adblock.adblock'
=========================== short test summary info ============================
ERROR tests/test_engine.py
ERROR tests/test_exceptions.py
ERROR tests/test_imports.py
ERROR tests/test_metadata.py
ERROR tests/test_redirect.py
ERROR tests/test_repr.py
```

File: contrib/python-adblock/template.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "cargo",
+    "python-build",
+    "python-installer",
+    "python-maturin",
+    "python-setuptools-rust",
+    "python-wheel",
+]
+makedepends = ["rust-std"]
+checkdepends = ["python-pytest-xdist"]
+pkgdesc = "Python wrapper for brave's adblocking library"
```
[the remark this answers is not in the document]
Comment (ttyyls) [reply contrib/python-adblock/template.py:15]: ok

File: contrib/python-adblock/template.py (hunk 1 above)
Comment (nekopsykose) [reply contrib/python-adblock/template.py:22]: yeah that happens when cwd has the name of `import x` in ./x but that folder doesn't have the built stuff, no way to work around it for now as we don't change cwd

File: contrib/python-adblock/template.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+maintainer = "ttyyls <contact@behri.org>"
+license = "Apache-2.0 OR MIT"
+url = "https://github.com/ArniDagur/python-adblock"
+source = f"{url}/archive/refs/tags/{pkgver}.tar.gz"
+sha256 = "cdde5db640f1d9205785641665b79e6edb7c51092a68536bc4907560c71136b2"
+# XXX: requires pre-installing the package
+options = ["!check"]
+
+
+def do_prepare(self):
```
Comment (nekopsykose) [inline contrib/python-adblock/template.py:25]:
```
def init_build(self):
    from cbuild.util import cargo

    renv = cargo.get_environment(self)
    self.make_env.update(renv)
```

**Pull request merged**